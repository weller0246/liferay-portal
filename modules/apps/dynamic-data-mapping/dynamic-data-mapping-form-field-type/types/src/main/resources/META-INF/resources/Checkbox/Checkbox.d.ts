/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import React from 'react';
declare const Main: React.FC<IProps>;
interface IProps extends ICheckboxProps {
	predefinedValue?: boolean | String[];
	readOnly?: boolean;
	showAsSwitcher?: boolean;
	showMaximumRepetitionsInfo?: boolean;
	systemSettingsURL: string;
	value?: boolean;
	visible?: boolean;
}
interface ICheckboxProps {
	checked: boolean;
	disabled?: boolean;
	label?: string;
	name: string;
	onChange: (event: {
		target: {
			value: boolean;
		};
	}) => void;
	required?: boolean;
	showLabel?: boolean;
}
export {Main};
export default Main;
