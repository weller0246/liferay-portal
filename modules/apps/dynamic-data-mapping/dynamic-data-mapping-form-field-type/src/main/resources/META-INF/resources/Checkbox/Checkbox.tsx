/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {ClayCheckbox, ClayInput} from '@clayui/form';
import ClayIcon from '@clayui/icon';
import React from 'react';

// @ts-ignore

import {FieldBase} from '../FieldBase/ReactFieldBase.es';

const Switcher: React.FC<ISwitcherProps> = ({
	checked,
	disabled,
	label,
	name,
	onChange,
	required,
	showLabel,
	showMaximumRepetitionsInfo,
	systemSettingsURL,
}) => {
	return (
		<>
			<label className="ddm-toggle-switch toggle-switch">
				<input
					checked={checked}
					className="toggle-switch-check"
					disabled={disabled}
					name={name}
					onChange={({target: {checked}}) => {
						onChange({target: {value: checked}});
					}}
					type="checkbox"
				/>

				<span aria-hidden="true" className="toggle-switch-bar">
					<span className="toggle-switch-handle"></span>

					{(showLabel || required) && (
						<span className="toggle-switch-text toggle-switch-text-right">
							{showLabel && label}

							{required && (
								<span className="ddm-label-required reference-mark">
									<ClayIcon symbol="asterisk" />
								</span>
							)}
						</span>
					)}
				</span>
			</label>
			{checked && showMaximumRepetitionsInfo && (
				<div className="ddm-info">
					<span className="ddm-tooltip">
						<ClayIcon symbol="info-circle" />
					</span>
					<div
						className="ddm-info-text"
						dangerouslySetInnerHTML={{
							__html: Liferay.Util.sub(
								Liferay.Language.get(
									'for-security-reasons-upload-field-repeatability-is-limited-the-limit-is-defined-in-x-system-settings-x'
								),
								`<a href=${systemSettingsURL} target="_blank">`,
								'</a>'
							),
						}}
					/>
				</div>
			)}
		</>
	);
};

const Checkbox: React.FC<ICheckboxProps> = ({
	checked,
	disabled,
	label,
	name,
	onChange,
	required,
	showLabel,
}) => {
	return (
		<ClayCheckbox
			checked={checked}
			disabled={disabled}
			label={showLabel ? label : ''}
			name={name}
			onChange={({target: {checked}}) => {
				onChange({target: {value: checked}});
			}}
		>
			{showLabel && required && (
				<span className="ddm-label-required reference-mark">
					<ClayIcon symbol="asterisk" />
				</span>
			)}
		</ClayCheckbox>
	);
};

const Main: React.FC<IProps> = ({
	label,
	name,
	onChange,
	predefinedValue,
	readOnly,
	required,
	showAsSwitcher = true,
	showLabel = true,
	showMaximumRepetitionsInfo = false,
	systemSettingsURL,
	value,
	visible,
	...otherProps
}) => {
	const Toggle = showAsSwitcher ? Switcher : Checkbox;

	const checked = !!(
		value ??
		(Array.isArray(predefinedValue)
			? predefinedValue[0] === 'true'
			: predefinedValue)
	);

	return (
		<FieldBase
			name={name}
			showLabel={false}
			visible={visible}
			{...otherProps}
		>
			<Toggle
				checked={checked}
				disabled={readOnly}
				label={label}
				name={name}
				onChange={onChange}
				required={required}
				showLabel={showLabel}
				showMaximumRepetitionsInfo={showMaximumRepetitionsInfo}
				systemSettingsURL={systemSettingsURL}
			/>
			<ClayInput name={name} type="hidden" value={`${checked}`} />
		</FieldBase>
	);
};

interface IProps extends ICheckboxProps {
	predefinedValue?: boolean | String[];
	readOnly?: boolean;
	showAsSwitcher?: boolean;
	showMaximumRepetitionsInfo?: boolean;
	systemSettingsURL: string;
	value?: boolean;
	visible?: boolean;
}
interface ISwitcherProps extends ICheckboxProps {
	showMaximumRepetitionsInfo: boolean;
	systemSettingsURL: string;
}

interface ICheckboxProps {
	checked: boolean;
	disabled?: boolean;
	label?: string;
	name: string;
	onChange: (event: {target: {value: boolean}}) => void;
	required?: boolean;
	showLabel?: boolean;
}

Main.displayName = 'Checkbox';

export {Main};
export default Main;
