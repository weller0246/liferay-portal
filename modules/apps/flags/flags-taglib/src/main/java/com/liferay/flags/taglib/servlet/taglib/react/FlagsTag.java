/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.flags.taglib.servlet.taglib.react;

import com.liferay.flags.taglib.internal.servlet.ServletContextUtil;
import com.liferay.flags.taglib.internal.servlet.taglib.util.FlagsTagUtil;
import com.liferay.portal.kernel.exception.PortalException;
import com.liferay.portal.kernel.json.JSONObject;
import com.liferay.portal.kernel.json.JSONUtil;
import com.liferay.portal.kernel.language.LanguageUtil;
import com.liferay.portal.kernel.log.Log;
import com.liferay.portal.kernel.log.LogFactoryUtil;
import com.liferay.portal.kernel.model.Company;
import com.liferay.portal.kernel.model.User;
import com.liferay.portal.kernel.theme.ThemeDisplay;
import com.liferay.portal.kernel.util.AggregateResourceBundle;
import com.liferay.portal.kernel.util.HashMapBuilder;
import com.liferay.portal.kernel.util.PortalUtil;
import com.liferay.portal.kernel.util.PortletKeys;
import com.liferay.portal.kernel.util.Validator;
import com.liferay.portal.kernel.util.WebKeys;
import com.liferay.portal.language.LanguageResources;
import com.liferay.taglib.util.IncludeTag;
import com.liferay.taglib.util.TagResourceBundleUtil;

import java.util.Map;
import java.util.ResourceBundle;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.jsp.PageContext;

/**
 * @author Ambrín Chaudhary
 * @author Jorge González
 */
public class FlagsTag extends IncludeTag {

	public String getClassName() {
		return _className;
	}

	public long getClassPK() {
		return _classPK;
	}

	public String getContentTitle() {
		return _contentTitle;
	}

	public String getContentURL() {
		return _contentURL;
	}

	public String getElementClasses() {
		return _elementClasses;
	}

	public String getMessage() {
		return _message;
	}

	public long getReportedUserId() {
		return _reportedUserId;
	}

	public boolean isEnabled() {
		return _enabled;
	}

	public boolean isLabel() {
		return _label;
	}

	public void setClassName(String className) {
		_className = className;
	}

	public void setClassPK(long classPK) {
		_classPK = classPK;
	}

	public void setContentTitle(String contentTitle) {
		_contentTitle = contentTitle;
	}

	public void setContentURL(String contentURL) {
		_contentURL = contentURL;
	}

	public void setElementClasses(String elementClasses) {
		_elementClasses = elementClasses;
	}

	public void setEnabled(boolean enabled) {
		_enabled = enabled;
	}

	public void setLabel(boolean label) {
		_label = label;
	}

	public void setMessage(String message) {
		_message = message;
	}

	@Override
	public void setPageContext(PageContext pageContext) {
		super.setPageContext(pageContext);

		setServletContext(ServletContextUtil.getServletContext());
	}

	public void setReportedUserId(long reportedUserId) {
		_reportedUserId = reportedUserId;
	}

	@Override
	protected void cleanUp() {
		super.cleanUp();

		_className = null;
		_classPK = 0;
		_contentTitle = null;
		_contentURL = null;
		_elementClasses = null;
		_enabled = true;
		_label = true;
		_message = null;
		_reportedUserId = 0;
	}

	@Override
	protected String getPage() {
		return _PAGE;
	}

	@Override
	protected void setAttributes(HttpServletRequest httpServletRequest) {
		try {
			String message = _getMessage();

			httpServletRequest.setAttribute(
				"liferay-flags:flags:data", _getData(message));

			httpServletRequest.setAttribute(
				"liferay-flags:flags:elementClasses", _elementClasses);
			httpServletRequest.setAttribute(
				"liferay-flags:flags:message", message);
			httpServletRequest.setAttribute(
				"liferay-flags:flags:onlyIcon", !_label);
		}
		catch (Exception exception) {
			_log.error(exception, exception);
		}
	}

	private Map<String, Object> _getData(String message)
		throws PortalException {

		return HashMapBuilder.<String, Object>put(
			"context",
			HashMapBuilder.<String, Object>put(
				"namespace", PortalUtil.getPortletNamespace(PortletKeys.FLAGS)
			).build()
		).put(
			"props",
			() -> {
				HttpServletRequest httpServletRequest = getRequest();

				ThemeDisplay themeDisplay =
					(ThemeDisplay)httpServletRequest.getAttribute(
						WebKeys.THEME_DISPLAY);

				return HashMapBuilder.<String, Object>put(
					"baseData", _getDataJSONObject(themeDisplay)
				).put(
					"captchaURI", FlagsTagUtil.getCaptchaURI(httpServletRequest)
				).put(
					"companyName",
					() -> {
						Company company = themeDisplay.getCompany();

						return company.getName();
					}
				).put(
					"disabled", !_enabled
				).put(
					"forceLogin", !FlagsTagUtil.isFlagsEnabled(themeDisplay)
				).put(
					"message",
					() -> {
						if (Validator.isNotNull(message)) {
							return message;
						}

						return null;
					}
				).put(
					"onlyIcon", !_label
				).put(
					"pathTermsOfUse",
					PortalUtil.getPathMain() + "/portal/terms_of_use"
				).put(
					"reasons",
					FlagsTagUtil.getReasons(
						themeDisplay.getCompanyId(), httpServletRequest)
				).put(
					"signedIn", themeDisplay.isSignedIn()
				).put(
					"uri", FlagsTagUtil.getURI(httpServletRequest)
				).build();
			}
		).build();
	}

	private JSONObject _getDataJSONObject(ThemeDisplay themeDisplay) {
		String namespace = PortalUtil.getPortletNamespace(PortletKeys.FLAGS);

		String contentURL = _contentURL;

		if (Validator.isNull(contentURL)) {
			contentURL = FlagsTagUtil.getCurrentURL(getRequest());
		}

		JSONObject dataJSONObject = JSONUtil.put(
			namespace + "className", _className
		).put(
			namespace + "classPK", _classPK
		).put(
			namespace + "contentTitle", _contentTitle
		).put(
			namespace + "contentURL", contentURL
		).put(
			namespace + "reportedUserId", _reportedUserId
		);

		if (themeDisplay.isSignedIn()) {
			User user = themeDisplay.getUser();

			dataJSONObject.put(
				namespace + "reporterEmailAddress", user.getEmailAddress());
		}

		return dataJSONObject;
	}

	private String _getMessage() {
		ResourceBundle resourceBundle = new AggregateResourceBundle(
			TagResourceBundleUtil.getResourceBundle(pageContext),
			LanguageResources.getResourceBundle(
				PortalUtil.getLocale(getRequest())));

		if (Validator.isNotNull(_message)) {
			return LanguageUtil.get(resourceBundle, _message);
		}

		return LanguageUtil.get(resourceBundle, "report");
	}

	private static final String _PAGE = "/flags/page.jsp";

	private static final Log _log = LogFactoryUtil.getLog(FlagsTag.class);

	private String _className;
	private long _classPK;
	private String _contentTitle;
	private String _contentURL;
	private String _elementClasses;
	private boolean _enabled = true;
	private boolean _label = true;
	private String _message;
	private long _reportedUserId;

}