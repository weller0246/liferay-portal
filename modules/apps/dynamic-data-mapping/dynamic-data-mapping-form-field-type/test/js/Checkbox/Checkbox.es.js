/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import '@testing-library/jest-dom/extend-expect';
import {cleanup, render} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import Checkbox from '../../../src/main/resources/META-INF/resources/Checkbox/Checkbox';

describe('Field Checkbox', () => {
	afterEach(cleanup);

	it('is not editable', () => {
		const {container} = render(<Checkbox readOnly />);

		expect(container).toMatchSnapshot();
	});

	it('has a helptext', () => {
		const {container} = render(<Checkbox tip="Type something" />);

		expect(container).toMatchSnapshot();
	});

	it('has an id', () => {
		const {container} = render(<Checkbox id="ID" />);

		expect(container).toMatchSnapshot();
	});

	it('has a label', () => {
		const {container} = render(<Checkbox label="label" />);

		expect(container).toMatchSnapshot();
	});

	it('has a predefined Value', () => {
		const {container} = render(<Checkbox placeholder="Option 1" />);

		expect(container).toMatchSnapshot();
	});

	it('is not required', () => {
		const {container} = render(<Checkbox required={false} />);

		expect(container).toMatchSnapshot();
	});

	it('is shown as a switcher', () => {
		const {container} = render(<Checkbox showAsSwitcher />);

		expect(container).toMatchSnapshot();
	});

	it('is shown as checkbox', () => {
		const {container} = render(<Checkbox showAsSwitcher={false} />);

		expect(container).toMatchSnapshot();
	});

	it('renders Label if showLabel is true', () => {
		const {container} = render(<Checkbox label showLabel />);

		expect(container).toMatchSnapshot();
	});

	it('has a value', () => {
		const {container} = render(<Checkbox value />);

		expect(container).toMatchSnapshot();
	});

	it('has a key', () => {
		const {container} = render(<Checkbox key="key" value />);

		expect(container).toMatchSnapshot();
	});

	it('call the onChange callback on the field change', () => {
		const handleFieldEdited = jest.fn();

		render(<Checkbox onChange={handleFieldEdited} />);

		userEvent.click(document.body.querySelector('input'));

		expect(handleFieldEdited).toHaveBeenCalled();
	});

	describe('Maximum Repetitions Info', () => {
		it('does not show the maximum repetitions info', () => {
			const {container} = render(<Checkbox value />);

			const ddmInfo = container.querySelector('.ddm-info');

			expect(ddmInfo).toBeNull();
		});

		it('does not show the maximum repetitions info if the value is false', () => {
			const {container} = render(
				<Checkbox showMaximumRepetitionsInfo value={false} />
			);

			const ddmInfo = container.querySelector('.ddm-info');

			expect(ddmInfo).toBeNull();
		});

		it('shows the maximum repetitions info', () => {
			const {container} = render(
				<Checkbox showMaximumRepetitionsInfo value />
			);

			const ddmInfo = container.querySelector('.ddm-info');

			expect(ddmInfo).not.toBeNull();
		});
	});

	describe('Boolean Field', () => {
		it('check if the boolean field is not checked if he has predefinedValue false', () => {
			const {queryByLabelText} = render(
				<Checkbox label="Boolean" predefinedValue={['false']} />
			);

			expect(queryByLabelText('Boolean')).not.toBeChecked();
		});

		it('check if the boolean field is checked if he has predefinedValue true', () => {
			const {queryByLabelText} = render(
				<Checkbox label="Boolean" predefinedValue={['true']} />
			);

			expect(queryByLabelText('Boolean')).toBeChecked();
		});

		it('check that with false predefinedValue the boolean field is checked when we enable it', () => {
			const onChange = jest.fn();

			const {queryByLabelText} = render(
				<Checkbox
					label="Boolean"
					onChange={onChange}
					predefinedValue={['false']}
				/>
			);

			const input = queryByLabelText('Boolean');
			userEvent.click(input);
			expect(onChange).toHaveBeenLastCalledWith({target: {value: true}});
		});

		it('check that with true predefinedValue the boolean field is not checked when we disabled it', () => {
			const onChange = jest.fn();
			const {queryByLabelText} = render(
				<Checkbox
					label="Boolean"
					onChange={onChange}
					predefinedValue={['true']}
				/>
			);

			const input = queryByLabelText('Boolean');
			userEvent.click(input);
			expect(onChange).toHaveBeenLastCalledWith({target: {value: false}});
		});

		it('check it shows the label when we set it up', () => {
			const {queryByLabelText} = render(
				<Checkbox label="Boolean" showLabel />
			);

			expect(queryByLabelText('Boolean')).toBeInTheDocument();
		});

		it('check if the required icon appears when the field is required', () => {
			render(<Checkbox required />);

			const requiredIcon = document.querySelector(
				'.lexicon-icon-asterisk'
			);

			expect(requiredIcon).toBeInTheDocument();
		});

		it('check if the required icon do not appears when the field is not required', () => {
			render(<Checkbox />);

			const requiredIcon = document.querySelector(
				'.lexicon-icon-asterisk'
			);

			expect(requiredIcon).not.toBeInTheDocument();
		});

		it('verify if the switcher do not appears when he is disabled in boolean field', () => {
			render(<Checkbox showAsSwitcher={false} />);

			const swithcerIcon = document.querySelector(
				'.toggle-switch-handle'
			);

			expect(swithcerIcon).not.toBeInTheDocument();
		});

		it('verify if the switcher appears when he is enabled in boolean field', () => {
			render(<Checkbox />);

			const swithcerIcon = document.querySelector(
				'.toggle-switch-handle'
			);

			expect(swithcerIcon).toBeInTheDocument();
		});
	});
});
