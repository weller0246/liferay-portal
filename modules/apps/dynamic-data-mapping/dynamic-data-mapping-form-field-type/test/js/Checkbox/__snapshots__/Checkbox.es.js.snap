// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Checkbox has a helptext 1`] = `
<div>
  <div
    class="form-group hide"
    tabindex="0"
  >
    <label
      class="ddm-toggle-switch toggle-switch"
    >
      <input
        class="toggle-switch-check"
        type="checkbox"
      />
      <span
        aria-hidden="true"
        class="toggle-switch-bar"
      >
        <span
          class="toggle-switch-handle"
        />
        <span
          class="toggle-switch-text toggle-switch-text-right"
        />
      </span>
    </label>
    <input
      class="form-control"
      type="hidden"
      value="false"
    />
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    <span
      aria-hidden="true"
      class="form-text"
    >
      Type something
    </span>
    <span
      class="sr-only"
    >
      Type something
      <br />
    </span>
  </div>
</div>
`;

exports[`Field Checkbox has a key 1`] = `
<div>
  <div
    class="form-group hide"
    tabindex="0"
  >
    <label
      class="ddm-toggle-switch toggle-switch"
    >
      <input
        checked=""
        class="toggle-switch-check"
        type="checkbox"
      />
      <span
        aria-hidden="true"
        class="toggle-switch-bar"
      >
        <span
          class="toggle-switch-handle"
        />
        <span
          class="toggle-switch-text toggle-switch-text-right"
        />
      </span>
    </label>
    <input
      class="form-control"
      type="hidden"
      value="true"
    />
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    
  </div>
</div>
`;

exports[`Field Checkbox has a label 1`] = `
<div>
  <div
    class="form-group hide"
    tabindex="0"
  >
    <label
      class="ddm-toggle-switch toggle-switch"
    >
      <input
        class="toggle-switch-check"
        type="checkbox"
      />
      <span
        aria-hidden="true"
        class="toggle-switch-bar"
      >
        <span
          class="toggle-switch-handle"
        />
        <span
          class="toggle-switch-text toggle-switch-text-right"
        >
          label
        </span>
      </span>
    </label>
    <input
      class="form-control"
      type="hidden"
      value="false"
    />
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    
  </div>
</div>
`;

exports[`Field Checkbox has a predefined Value 1`] = `
<div>
  <div
    class="form-group hide"
    tabindex="0"
  >
    <label
      class="ddm-toggle-switch toggle-switch"
    >
      <input
        class="toggle-switch-check"
        type="checkbox"
      />
      <span
        aria-hidden="true"
        class="toggle-switch-bar"
      >
        <span
          class="toggle-switch-handle"
        />
        <span
          class="toggle-switch-text toggle-switch-text-right"
        />
      </span>
    </label>
    <input
      class="form-control"
      type="hidden"
      value="false"
    />
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    
  </div>
</div>
`;

exports[`Field Checkbox has a value 1`] = `
<div>
  <div
    class="form-group hide"
    tabindex="0"
  >
    <label
      class="ddm-toggle-switch toggle-switch"
    >
      <input
        checked=""
        class="toggle-switch-check"
        type="checkbox"
      />
      <span
        aria-hidden="true"
        class="toggle-switch-bar"
      >
        <span
          class="toggle-switch-handle"
        />
        <span
          class="toggle-switch-text toggle-switch-text-right"
        />
      </span>
    </label>
    <input
      class="form-control"
      type="hidden"
      value="true"
    />
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    
  </div>
</div>
`;

exports[`Field Checkbox has an id 1`] = `
<div>
  <div
    aria-labelledby="ID"
    class="form-group hide"
    tabindex="0"
  >
    <label
      class="ddm-toggle-switch toggle-switch"
    >
      <input
        class="toggle-switch-check"
        type="checkbox"
      />
      <span
        aria-hidden="true"
        class="toggle-switch-bar"
      >
        <span
          class="toggle-switch-handle"
        />
        <span
          class="toggle-switch-text toggle-switch-text-right"
        />
      </span>
    </label>
    <input
      class="form-control"
      type="hidden"
      value="false"
    />
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    
  </div>
</div>
`;

exports[`Field Checkbox is not editable 1`] = `
<div>
  <div
    class="form-group hide"
    tabindex="0"
  >
    <label
      class="ddm-toggle-switch toggle-switch"
    >
      <input
        class="toggle-switch-check"
        disabled=""
        type="checkbox"
      />
      <span
        aria-hidden="true"
        class="toggle-switch-bar"
      >
        <span
          class="toggle-switch-handle"
        />
        <span
          class="toggle-switch-text toggle-switch-text-right"
        />
      </span>
    </label>
    <input
      class="form-control"
      type="hidden"
      value="false"
    />
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    
  </div>
</div>
`;

exports[`Field Checkbox is not required 1`] = `
<div>
  <div
    class="form-group hide"
    tabindex="0"
  >
    <label
      class="ddm-toggle-switch toggle-switch"
    >
      <input
        class="toggle-switch-check"
        type="checkbox"
      />
      <span
        aria-hidden="true"
        class="toggle-switch-bar"
      >
        <span
          class="toggle-switch-handle"
        />
        <span
          class="toggle-switch-text toggle-switch-text-right"
        />
      </span>
    </label>
    <input
      class="form-control"
      type="hidden"
      value="false"
    />
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    
  </div>
</div>
`;

exports[`Field Checkbox is shown as a switcher 1`] = `
<div>
  <div
    class="form-group hide"
    tabindex="0"
  >
    <label
      class="ddm-toggle-switch toggle-switch"
    >
      <input
        class="toggle-switch-check"
        type="checkbox"
      />
      <span
        aria-hidden="true"
        class="toggle-switch-bar"
      >
        <span
          class="toggle-switch-handle"
        />
        <span
          class="toggle-switch-text toggle-switch-text-right"
        />
      </span>
    </label>
    <input
      class="form-control"
      type="hidden"
      value="false"
    />
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    
  </div>
</div>
`;

exports[`Field Checkbox is shown as checkbox 1`] = `
<div>
  <div
    class="form-group hide"
    tabindex="0"
  >
    <div
      class="custom-control custom-checkbox"
    >
      <label>
        <input
          class="custom-control-input"
          type="checkbox"
        />
        <span
          class="custom-control-label"
        />
      </label>
    </div>
    <input
      class="form-control"
      type="hidden"
      value="false"
    />
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    
  </div>
</div>
`;

exports[`Field Checkbox renders Label if showLabel is true 1`] = `
<div>
  <div
    class="form-group hide"
    tabindex="0"
  >
    <label
      class="ddm-toggle-switch toggle-switch"
    >
      <input
        class="toggle-switch-check"
        type="checkbox"
      />
      <span
        aria-hidden="true"
        class="toggle-switch-bar"
      >
        <span
          class="toggle-switch-handle"
        />
        <span
          class="toggle-switch-text toggle-switch-text-right"
        />
      </span>
    </label>
    <input
      class="form-control"
      type="hidden"
      value="false"
    />
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    
  </div>
</div>
`;
